### PR TITLE
456 more collaborations

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Collaborations now possible between all active users
- View only on initial collaboration creation
- Both view and edit are only allowed if reciprocal.
- New collaborations/edit_request REST API to upgrade a Collaboration to Edit
- New notification generated to the user to inform them of this

---

**Database Updates **

Collaboration association has new boolean flag to indicate if the user initiated the collaboration

**REST API Updates **

```
[POST] /api/v1/collaborations/edit_request/{collaboration_guid}
No payload. This is how the user would request that the the collaboration is upgraded to an edit one
```

### CODE REVIEW

File: tests/modules/collaborations/resources/test_collaboration_notification.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=missing-docstring
+import tests.modules.collaborations.resources.utils as collab_utils
+import tests.modules.notifications.resources.utils as notif_utils
+from tests import utils as test_utils
+
+
+# Full sequence, create the collaboration, validate that the notification is received, approve the view aspect
+# Request edit, validate that notification is received, approve edit.
+def test_edit_collaboration(flask_app_client, researcher_1, researcher_2, db, request):
```
Comment (hwindsor) [inline tests/modules/collaborations/resources/test_collaboration_notification.py:10]: @brmscheiner I created a test that did the full sequence as you asked

Comment (brmscheiner) [reply tests/modules/collaborations/resources/test_collaboration_notification.py:10]: thanks for doing this!

**Review by colinwkingen — APPROVED** [review]
Looks good, also well tested.

File: app/modules/collaborations/resources.py [hunk 2]
```diff
@@ -81,8 +80,8 @@ def post(self, args):
         if not other_user:
             abort(400, f'User with guid {other_user_guid} not found')
 
-        if not other_user.is_researcher:
-            abort(400, f'User with guid {other_user_guid} is not a researcher')
+        if not other_user.is_active:
+            abort(400, f'User with guid {other_user_guid} is not active')
```
[the remark this answers is not in the document]
Comment (hwindsor) [reply app/modules/collaborations/resources.py:85]: Valid question. Certainly outside of the scope of this PR. Added as a question to knucklehead doc.

**Pull request merged**